ext/nnstreamer/tensor_filter/tensor_filter_llamacpp.cc

```diff
@@ -118,24 +123,68 @@ TensorFilterLlamaCpp::getEmptyInstance ()
   return *(new TensorFilterLlamaCpp ());
 }
 
+/**
+ * @brief Parse custom prop and set instance options accordingly.
+ */
+void
+TensorFilterLlamaCpp::parseCustomProperties (const GstTensorFilterProperties *prop)
+{
+  guint len, i;
+  gchar **options;
+
+  if (!prop->custom_properties)
+    return;
+
+  options = g_strsplit (prop->custom_properties, ",", -1);
+  len = g_strv_length (options);
+
+  for (i = 0; i < len; ++i) {
+    gchar **option = g_strsplit (options[i], ":", -1);
+
+    if (g_strv_length (option) > 1) {
+      g_strstrip (option[0]);
+      g_strstrip (option[1]);
+
+      if (g_ascii_strcasecmp (option[0], "num_gpu_layers") == 0) {
+        n_gpu_layers = (int) g_ascii_strtoull (option[1], NULL, 10);
+      } else if (g_ascii_strcasecmp (option[0], "num_predict") == 0) {
+        n_predict = (int) g_ascii_strtoull (option[1], NULL, 10);
+      } else {
+        throw std::invalid_argument (
```

Comment:
g_strfreev before throw.

Reply:
Updated it to use `unique_ptr`
Thank you :)